### PR TITLE
Address security vulnerabilities

### PR DESCRIPTION
**Pull request by Tawakalt**
**Proposed changes**:
- Fixes [ATO-1497](https://rasahq.atlassian.net/browse/ATO-1497)

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [X] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-1497]: https://rasahq.atlassian.net/browse/ATO-1497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -193,7 +195,7 @@ version = ">=1.4.1,<1.7.3"
 python = "~=3.7.0"
 
 [[tool.poetry.dependencies.scipy]]
-version = ">=1.4.1,<1.9.0"
+version = ">=1.4.1,<1.8.1"
```
Comment (sanchariGr) [inline pyproject.toml:198]: The fixed version seems to be 1.10.0 https://storage.googleapis.com/rasa-container-scan-reports/rasa-plus-dependency-report-3.6.6-20230828.txt

Comment (Tawakalt) [reply pyproject.toml:198]: I wasn't sure if I should reduce the range from a wider  `>=1.4.1,<1.9.0` to just `>=1.10.0`?
Is that better than reducing the range to versions before the vulnerability? `version = ">=1.4.1,<1.8.1"`

Comment (sanchariGr) [reply pyproject.toml:198]: hmm not sure, @rasa-jmac do you have a suggestion here?

Comment (rasa-jmac) [reply pyproject.toml:198]: It's always better to move to at least the fixed version, which is indeed `>=1.10.0`

I don't see any docs that say the vulnerability _wasn't_ present in <1.8.1, just that that is the vulnerable version we have installed. Indeed, the [vuln listing](https://nvd.nist.gov/vuln/detail/CVE-2023-25399) (see "Known Affected Software Configurations") suggests it is every release before 1.10.0.

Also, even if docs say a version isn't present in an older version, we don't really want to be shipping older versions of software as a fix for things. We wouldn't be able to say for sure if the vuln really isn't present, or if the maintainers just didn't check back that far to see if it was.

Comment (Tawakalt) [reply pyproject.toml:198]: That makes sense. Thanks for your input!

Comment (rasa-jmac) [reply pyproject.toml]: Also, by moving backwards to older versions, we're more likely to reintroduce older vulns that have already previously been fixed, and actually causing a larger problem than the one we're trying to fix.

**Review by sanchariGr — COMMENTED** [review]
🚀

**Review by sanchariGr — APPROVED** [review]

**Pull request merged**